Go version is pinned to 1.12.4

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Running `make docker-build` or `make binaries` from an environment using the latest go version (`1.12.6`) returns `compile: version "go1.12.6" does not match go tool version "go1.12.4"`.

**What did you expect to happen:**
I would expect to be able to build using any 1.12 go version (in fact https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/development.md states "Get the latest patch version for go v1.12." as a prerequisite).

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- cluster-api-provider-azure version: master
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

bumping the version in `WORKSPACE` results in: 

```
make reset-bazel && make test           ✔  6509  17:06:03
bazel clean --expunge
INFO: Starting clean.
GOPATH=/Users/cerobert/go bazel run //:generate
Starting local Bazel server and connecting to it...
ERROR: /Users/cerobert/go/src/sigs.k8s.io/cluster-api-provider-azure/WORKSPACE:47:1: Traceback (most recent call last):
	File "/Users/cerobert/go/src/sigs.k8s.io/cluster-api-provider-azure/WORKSPACE", line 47
		go_register_toolchains(go_version = "1.12.6")
	File "/private/var/tmp/_bazel_cerobert/2b6c8f6bcc2c3174b6bcb7a811b8ef9f/external/io_bazel_rules_go/go/toolchain/toolchains.bzl", line 362, in go_register_toolchains
		fail("Unknown go version {}".format(g...))
Unknown go version 1.12.6
ERROR: Error evaluating WORKSPACE file
ERROR: error loading package '': in /private/var/tmp/_bazel_cerobert/2b6c8f6bcc2c3174b6bcb7a811b8ef9f/external/io_bazel_rules_docker/container/container.bzl: in /private/var/tmp/_bazel_cerobert/2b6c8f6bcc2c3174b6bcb7a811b8ef9f/external/io_bazel_rules_docker/container/bundle.bzl: Encountered error while reading extension file 'lib/dicts.bzl': no such package '@bazel_skylib//lib': error loading package 'external': Could not load //external package
ERROR: error loading package '': in /private/var/tmp/_bazel_cerobert/2b6c8f6bcc2c3174b6bcb7a811b8ef9f/external/io_bazel_rules_docker/container/container.bzl: in /private/var/tmp/_bazel_cerobert/2b6c8f6bcc2c3174b6bcb7a811b8ef9f/external/io_bazel_rules_docker/container/bundle.bzl: Encountered error while reading extension file 'lib/dicts.bzl': no such package '@bazel_skylib//lib': error loading package 'external': Could not load //external package
INFO: Elapsed time: 3.159s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
make: *** [generate] Error 1
```

I also tried changing it to 
```go_register_toolchains(
    go_version = "1.12",
)```

in which case `make reset-bazel && make test` returns no errors but then running `make test` returns `compile: version "go1.12.6" does not match go tool version "go1.12"` ( `compile: version "go1.12.4" does not match go tool version "go1.12"` now that I have 1.12.4).

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -16,8 +16,11 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "86ae934bd4c43b99893fc64be9d9fc684b81461581df7ea8fc291c816f5ee8c5",
-    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.18.3/rules_go-0.18.3.tar.gz"],
+    urls = [
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/0.18.6/rules_go-0.18.6.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/0.18.6/rules_go-0.18.6.tar.gz",
+    ],
+    sha256 = "f04d2373bcaf8aa09bccb08a98a57e721306c8f6043a2a0ee610fd6853dcde3d",
 )
 
 http_archive(
@@ -45,7 +48,7 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 go_rules_dependencies()
 
 go_register_toolchains(
-    go_version = "1.12.4",
+    go_version = "1.12.6",
 )
 
 load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")

